test/infrastructure/docker/internal/controllers/backends/docker/dockermachine_backend.go

```diff
@@ -362,22 +399,13 @@ func (r *MachineBackendReconciler) ReconcileDelete(ctx context.Context, cluster
 	// this visible to the users.
 	// NB. The operation in docker is fast, so there is the chance the user will not notice the status change;
 	// nevertheless we are issuing a patch so we can test a pattern that will be used by other providers as well
-	newPatchHelperFunc := r.NewPatchHelperFunc
-	if newPatchHelperFunc == nil {
-		newPatchHelperFunc = patch.NewHelper
-	}
-	patchHelper, err := newPatchHelperFunc(dockerMachine, r.Client)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-	conditions.MarkFalse(dockerMachine, infrav1.ContainerProvisionedCondition, clusterv1.DeletingReason, clusterv1.ConditionSeverityInfo, "")
-
-	patchDevMachineFunc := r.PatchDevMachineFunc
-	if patchDevMachineFunc == nil {
-		patchDevMachineFunc = r.PatchDevMachine
-	}
-	if err := patchDevMachineFunc(ctx, patchHelper, dockerMachine, util.IsControlPlaneMachine(machine)); err != nil {
-		return ctrl.Result{}, errors.Wrap(err, "failed to patch DockerMachine")
+	if conditions.GetReason(dockerMachine, infrav1.ContainerProvisionedCondition) != clusterv1.DeletingReason {
```

Comment:
I wonder if it would be better to check for status false here instead of reason != ""

Reply:
The goal of this if is to surface deleting is started, and we should do this if the condition is reporting something different than deleting

Reply:
I think I misplaced this comment